mixins/common.yml

```
@@ -1058,24 +1058,59 @@ actions:
     - setGlobalRepoRootUrl
     - fetchS3LogsBucketInfos
     - fetchRcloneFiles
+    - setupLogsRCloneConfig:
+        provider: ${globals.logs_bucket_provider}
+        region: ${globals.s3_bucket_region}
+        __secret__access_key: ${globals.__secret__access_key}
+        __secret__secret_key: ${globals.__secret__secret_key}
+        __secret__aes256_key: ${globals.__secret__aes256_key}
 
   fetchRcloneFiles:
     # Parameters
     # target: the node/nodeGroup to setup
     - cmd[*]: |-
-        __secret__access_key="${globals.__secret__access_key}"
-        __secret__secret_key="${globals.__secret__secret_key}"
-        region="${globals.s3_bucket_region}"
-        rclone_conf=/root/.config/rclone/rclone.conf
+        rclone_dir=/root/.config/rclone
+        rclone_conf=$rclone_dir/rclone.conf
         curl -fSsLo /etc/cron.d/rclone-cron ${globals.repoRootUrl}/assets/common/rclone-cron
-        mkdir -p /root/.config/rclone
+        mkdir -p $rclone_dir
         curl -fSsLo $rclone_conf ${globals.repoRootUrl}/assets/common/rclone.conf
         curl -fSsLo /usr/local/bin/rclone-logs.sh ${globals.repoRootUrl}/assets/common/rclone-logs.sh
         chmod +x /usr/local/bin/rclone-logs.sh
+      user: root
+
+  setupLogsRCloneConfig:
+    # Parameters
+    #   - provider: the cloud provider hosting the s3 bucket
+    #   - region: the region of the s3 bucket
+    #   - __secret__access_key: the access key used by rclone
+    #   - __secret__secret_key: the secret key used by rclone
+    #   - __secret__aes256_key: the aes256 key used by rclone to encrypt/decrypt data when region is OVH (optional)
+    - cmd[*]: |-
+        rclone_conf=/root/.config/rclone/rclone.conf
+        provider="${this.provider}"
+        region="${this.region}"
+        __secret__access_key="${this.__secret__access_key}"
+        __secret__secret_key="${this.__secret__secret_key}"
+        __secret__aes256_key="${this.__secret__aes256_key.print()}"
+
+        set -e
+        sed -i -e "s/LOGS_REGION_PLACEHOLDER/$region/" \
+               -e "s/LOGS_ACCESS_KEY_PLACEHOLDER/$__secret__access_key/" \
+               -e "s#LOGS_SECRET_KEY_PLACEHOLDER#$__secret__secret_key#" $rclone_conf
 
-        sed -i "s/LOGS_ACCESS_KEY_PLACEHOLDER/$__secret__access_key/" $rclone_conf
-        sed -i "s/LOGS_SECRET_KEY_PLACEHOLDER/$__secret__secret_key/" $rclone_conf
-        sed -i "s/LOGS_REGION_PLACEHOLDER/$region/" $rclone_conf
+        if [ "$provider" = "ovh" ]; then
+          provider="Other"
+          endpoint="https://s3.$region.io.cloud.ovh.net/"
+          sed -i -e "s;LOGS_ENDPOINT_PLACEHOLDER;$endpoint;" \
+                 -e  "s#LOGS_AES256_KEY_PLACEHOLDER#$__secret__aes256_key#" \
+                 -e  -e "/#PROVIDER_AWS/d" \
+                 -e "s/ #PROVIDER_OVH//" $rclone_conf
+        else
+          provider=AWS
+          sed -i "/#PROVIDER_OVH/d" $rclone_conf \
+                 -e "s/ #PROVIDER_AWS//"
```

```suggestion
          sed -i -e "/#PROVIDER_OVH/d" \
                 -e "s/ #PROVIDER_AWS//" $rclone_conf
```
(to stick to the spirit, although personally I prefer the `-i <file>` in the first or last line)

---

Done!